docs/source/reference/ha.rst

```diff
@@ -0,0 +1,202 @@
+
+n st2api processes can be behind a loadbalancer in an active-active configuration. Each of these
+processes can be deployed on separate compute instances.
+
+st2auth
+~~~~~~~
+All authentication is managed by this process. This process needs connection to MongoDB and an authentication
+backend. See `Authentication backends<ref-sensors-authoring-a-sensor>` for more information. It is a gunicorn
+managed process which by default listens on port ``9102`` and is front-ended by Nginx acting as a reverse proxy.
+
+n st2auth processes can be behind a loadbalancer in an active-active configuration. Each of these processes
+can be deployed on separate compute instances. If using the PAM authentication backend special care has to be
+taken to guarantee that all boxes on which an instance of st2auth runs should have the same users. Generally,
+all st2auth process should see the same identities, via some provider if applicable, for the system to work
+predictably.
+
+st2stream
+~~~~~~~~~
+This process exposes a server-sent event stream. Requires access to both MongoDB and RabbitMq. It is also a
+gunicorn managed process, listens on port ``9102`` by default and is front-ended by Nginx acting as a reverse
+proxy. Clients like WebUI and ChatOps maintain a persistent connection with an st2stream process and receive
+update from the st2stream server.
+
+n st2stream process can be behind a loadbalance in an active-active configuration. Since clients maintain
+a persistent connection with a specific instance the client will loose event momentarily if an st2stream
+process goes down. It would be the responsibility of the client to reconnect to an alternate stream connection
+via the loadbalancer.
+
+st2sensorcontainer
+~~~~~~~~~~~~~~~~~~
+st2sensorcontainer manages the sensors to be run on a node. It will start, stop and restart based on policy
+sensor running on a node. In this case node is same as a compute instance i.e. Virtual Machine and somewhere
+down the line could be a container.
+
+It is possible to run a st2sensorcontainer in HA mode by running one on each compute instance thus leading
+to various sensor nodes. Each sensor node needs to be provided with proper partition information to share work
+with other sensor nodes so that the same sensor does not run on different nodes. See :doc:`/reference/sensor_partitioning`
+for information on how to partition sensors. At this point st2sensorcontainer's do not form a cluster and
+distribute work or take over new work if some nodes in the cluster disappear. It would however be possible to a
+sensor itself to be implemented with HA in mind so that the same sensor can be deployed on multiple nodes with
+the sensor managing active-active or active-passive. Providing some platform level HA support for sensors is
+likely to be an enhancement to StackStorm in future releases.
+
+
+st2rulesengine
+~~~~~~~~~~~~~~
+This is a dual purpose process - its main function is to evaluate rules when it sees TriggerInstances and
```

Comment:
I'm concerned on maintaining two sets of definitions of services, here and in the /install/overview. Most of the description here is color on how it operates in multi-node, but some give pretty good definitions of the services themselves. Can we keep the separation (definitions in overview, HA details/roles here)? If it's not easy, keep as is, but also put some good soundbites back to overview, too.